mlflow/prophet.py

```diff
@@ -0,0 +1,296 @@
+):
+    """
+    Save a Prophet model to a path on the local file system.
+
+    :param pr_model: Prophet model (an instance of Prophet() after having :func:`.fit()`
+                     applied to be saved.
+    :param path: Local path where the serialized model (as JSON) is to be saved.
+    :param conda_env: {{ conda_env }}
+    :param mlflow_model: :py:mod:`mlflow.models.Model` this flavor is being added to.
+    :param signature: :py:class:`ModelSignature <mlflow.models.ModelSignature>`
+                      describes model input and output :py:class:`Schema <mlflow.types.Schema>`.
+                      The model signature can be :py:func:`inferred <mlflow.models.infer_signature>`
+                      from datasets with valid model input (e.g. the training dataset with target
+                      column omitted) and valid model output (e.g. model predictions generated on
+                      the training dataset), for example:
+
+                      .. code-block:: python
+
+                        from mlflow.models.signature import infer_signature
+
+                        model = Prophet().fit(df)
+                        train = model.history
+                        predictions = model.predict(model.make_future_dataframe(30))
+                        signature = infer_signature(train, predictions)
+    :param input_example: Input example provides one or several instances of valid
+                          model input. The example can be used as a hint of what data to feed the
+                          model. The given example will be converted to a Pandas DataFrame and then
+                          serialized to json using the Pandas split-oriented format. Bytes are
+                          base64-encoded.
+    :param pip_requirements: {{ pip_requirements }}
+    :param extra_pip_requirements: {{ extra_pip_requirements }}
+    """
+
+    import prophet
+    import pystan
+    import numpy
+
+    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+
+    path = os.path.abspath(path)
+    if os.path.exists(path):
+        raise MlflowException(f"Path '{path}' already exists")
+    os.makedirs(path)
+    if mlflow_model is None:
+        mlflow_model = Model()
+    if signature is not None:
+        mlflow_model.signature = signature
+    if input_example is not None:
+        _save_example(mlflow_model, input_example, path)
+
+    model_data_path = os.path.join(path, _MODEL_BINARY_FILE_NAME)
+    _save_model(pr_model, model_data_path)
+
+    model_bin_kwargs = {_MODEL_BINARY_KEY: _MODEL_BINARY_FILE_NAME}
+    pyfunc.add_to_model(
+        mlflow_model, loader_module="mlflow.prophet", env=_CONDA_ENV_FILE_NAME, **model_bin_kwargs
+    )
+    flavor_conf = {
+        _MODEL_TYPE_KEY: pr_model.__class__.__name__,
+        _SAVE_FORMAT_KEY: kwargs.get("format", "pr"),
+        **model_bin_kwargs,
+    }
+    mlflow_model.add_flavor(
+        FLAVOR_NAME,
+        numpy_version=numpy.__version__,
```

Comment:
Is there something special about the numpy version that may impact prophet behavior? If so, should we also include it in the list of generated pip requirements? If not, can we drop it from the flavor conf?

Reply:
Lingering debugging red herring ;) removed.